### DISCUSSION TITLE
contact modify together with constraint collision not working

### DISCUSSION THREAD

**Discussion by yananZZZZ** [NVIDIA-Omniverse/PhysX#148] [Q&A] [answered]
Hi,

  I try to implement the effect that a conveyor belt that move vertically by a sliding joint can see the attachment. What's I have done is just used contact modify callback to set the targetvelocity for those contact pair to make them the same direction as my conveyor. But I found out that my obstacle drop right through my conveyor, it looks like a contact solver fails to handle penetration after contact modification.
  I just want to know if there is some tips or limitation of contact solver and contact modify.

thank you.
![conveyor](https://github.com/NVIDIA-Omniverse/PhysX/assets/122350362/589bab51-9093-4189-84eb-3576ce03767e)
![ConveyorDrop](https://github.com/NVIDIA-Omniverse/PhysX/assets/122350362/fb94d69f-80cd-476b-80cd-f0dfce0cd833)

**Comment by preist-nvidia** [comment]
So top is implemented using a sliding distance joint?

The contact modification approach should work - this is how we implement conveyor belts in Omniverse Physics as well. You are just setting a target velocity using `PxContactSet::setTargetVelocity`?

**Reply by yananZZZZ** [reply, marked as answer]
Hi

I found the root cause eventually. it because I use the setDominanceGroup function wrongly. After I set the correct dominance group, every works prefectly.

thank you
